### PR TITLE
[SYCL-MLIR]: Add sycl-mlir test to list of test run automatically on github

### PR DESCRIPTION
**Pull request by etiotto**
This PR adds the sycl-mlir unit tests to the list of tests that should be run before a PR can be merged.

### CODE REVIEW

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 1]
```diff
@@ -110,6 +110,18 @@ jobs:
...
+        cmake --build $GITHUB_WORKSPACE/build --target check-mlir-sycl
+    - name: check-cgeist
+      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
+      run: |
+        cmake --build $GITHUB_WORKSPACE/build --target check-cgeist
+    - name: check-polygeist-opt
+      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
+      run: |
+        cmake --build $GITHUB_WORKSPACE/build --target check-polygeist-opt
     - name: check-sycl
```
Comment (bader) [inline .github/workflows/sycl_linux_build_and_test.yml:125]: Don't you want to remove existing checks? I suppose they might be not relevant (at least for now).

**Review by bader — DISMISSED** [review]

**Comment by bader** [comment]
The script is trying to merge `sycl` branch head before testing. We need to change it to `sycl-mlir`.

File: .github/workflows/sycl_linux_build_and_test.yml (hunk 1 above)
Comment (Naghasan) [reply .github/workflows/sycl_linux_build_and_test.yml:125]: I think they would become relevant quickly enough

**Comment by pvchupin** [comment]
@etiotto, it makes sense to make sure at least some testing passes first, before extending it. Currently it fails on checkout. I think we may have some hardcodes to sycl branch.
Also what testing volume you have in mind? What would be enough to run pre-commit for this project?

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 2]
```diff
@@ -110,6 +110,18 @@ jobs:
       run: |
         export XDG_CACHE_HOME=$GITHUB_WORKSPACE/os_cache
         cmake --build $GITHUB_WORKSPACE/build --target check-clang
+    - name: check-mlir-sycl
+      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
+      run: |
+        cmake --build $GITHUB_WORKSPACE/build --target check-mlir-sycl
```
Comment (Naghasan) [inline .github/workflows/sycl_linux_build_and_test.yml:116]:
```suggestion
    - name: check-mlir
      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
      run: |
        cmake --build $GITHUB_WORKSPACE/build --target check-mlir
    - name: check-mlir-sycl
      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
      run: |
        cmake --build $GITHUB_WORKSPACE/build --target check-mlir-sycl
```

not directly needed but depends on it, so might be better to have it checked

**Comment by bader** [comment]
Try adding `default_branch: $GITHUB_BASE_REF` to https://github.com/intel/llvm/blob/sycl-mlir/.github/workflows/sycl_linux_build_and_test.yml after line 78.

File: .github/workflows/sycl_linux_build_and_test.yml (hunk 1 above)
Comment (bader) [reply .github/workflows/sycl_linux_build_and_test.yml:125]: Okay. We need to check that we have enough CI resources to handle extended pre-commit for another branch.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -64,3 +64,7 @@ llvm/lib/SYCLLowerIR/CMakeLists.txt @intel/dpcpp-tools-reviewers @intel/dpcpp-es
 .github/workflows/ @intel/dpcpp-devops-reviewers
 buildbot/ @intel/dpcpp-devops-reviewers
 devops/ @intel/dpcpp-devops-reviewers
+
+# SYCL MLIR
+mlir-sycl/ @etiotto @whitneywhtsang
```
Comment (etiotto) [inline .github/CODEOWNERS]: will add also Victor Lomuller (@Naghasan) here.

File: .github/workflows/sycl_linux_build_and_test.yml (hunk 2 above)
Comment (etiotto) [reply .github/workflows/sycl_linux_build_and_test.yml:121]: Agree. I will add a step to check that mlir tests are clean.

**Comment by etiotto** [comment]
> @etiotto, it makes sense to make sure at least some testing passes first, before extending it. Currently it fails on checkout. I think we may have some hardcodes to sycl branch. Also what testing volume you have in mind? What would be enough to run pre-commit for this project?

We could rely on the testing done in the `sycl` branch and for this branch `sycl-mlir` we could run only additional tests, namely:  
```
   ninja check-mlir; ninja check-mlir-sycl; ninja check cgeist; ninja check-polygeist-opt
```
This would minimize the load on existing resources. @Naghasan would that be reasonable?

**Comment by bader** [comment]
Hm... for some reason DEFAUL_BRANCH variable is still `sycl`. Probably, we can hard-code `sycl-mlir` [here](https://github.com/intel/llvm/blob/sycl-mlir/devops/actions/cached_checkout/action.yml#L18) and open an issue to make cached_checkout action configurable.

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 4]
```diff
@@ -76,6 +76,7 @@ jobs:
         path: src
         ref: ${{ inputs.build_ref || github.ref }}
         cache_path: "/__w/repo_cache/"
+        default_branch: $GITHUB_BASE_REF
```
Comment (pvchupin) [inline .github/workflows/sycl_linux_build_and_test.yml:79]:
```suggestion
        default_branch: sycl-mlir
```
Can we try this?

Comment (etiotto) [reply .github/workflows/sycl_linux_build_and_test.yml:79]: yes will try it

**Comment by pvchupin** [comment]
I think it's actually the same problem with the pull_request_target that we experienced already.
Basically in the pre-commit testing of this specific PR the old version (base) of workload is picked up, not the changed one.

We need to commit this and test in other new PR.

**Comment by etiotto** [comment]
With ac6b7a630c0e4c76e46479ca20f261d3f31b2382 the lint work is now using the `sycl-mlir` branch:

 
image

**Review by pvchupin — APPROVED** [review]
We can try to rollback to the original patch suggested by @bader and commit, then test with separate PR.

**Comment by pvchupin** [comment]
> With [ac6b7a6](https://github.com/intel/llvm/commit/ac6b7a630c0e4c76e46479ca20f261d3f31b2382) the lint work is now using the `sycl-mlir` branch:
> 
> image

But we're still on
```
2022-09-08T19:37:37.2495324Z with:
2022-09-08T19:37:37.2496029Z   default_branch: sycl
```
though on Build + LIT

**Comment by etiotto** [comment]
I wonder whether we need to commit this PR to make the script changes and see if the next PR is OK @pvchupin

**Comment by etiotto** [comment]
The LIT job attempts to compare this changes with the sycl branch:

image

**Comment by etiotto** [comment]
Committing as suggested in https://github.com/intel/llvm/pull/6738#issuecomment-1241153627

**Pull request merged**